src/program.cpp

```diff
@@ -445,6 +445,30 @@ bool spir_binary::parse_pushconstant(const std::vector<std::string>& tokens,
     return true;
 }
 
+bool spir_binary::parse_specconstant(const std::vector<std::string>& tokens,
+                                     int toknum) {
+    auto name = tokens[toknum++];
+    spec_constant constant;
+    if (name == "workgroup_size_x") {
```

Comment:
We may want to turn this into a map lookup in the future to make things more compact.